js/src/views/Signer/reducers/requests.js

```diff
@@ -108,3 +108,8 @@ function setIsSending (pending, id, isSending) {
     return p;
   }).slice();
 }
+
+function addTimestamp (request) {
+  request.date = new Date();
```

Comment:
This modifies the `request` object and can therefore later have unintended side effects. I'd propose `(r) => {request: r, timestamp: new Date()}`. If you don't need the request, `(r) => new Date()`.